pkg/karmadactl/top/top_pods.go

```diff
@@ -0,0 +1,274 @@
+		Example:               topPodExample,
+		ValidArgsFunction:     completion.ResourceNameCompletionFunc(f, "pod"),
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(o.Complete(f, cmd, args))
+			cmdutil.CheckErr(o.Validate())
+			cmdutil.CheckErr(o.RunTopPod())
+		},
+		Aliases: []string{"pods", "po"},
+	}
+	cmdutil.AddLabelSelectorFlagVar(cmd, &o.LabelSelector)
+	options.AddKubeConfigFlags(cmd.Flags())
+	cmd.Flags().StringVarP(options.DefaultConfigFlags.Namespace, "namespace", "n", *options.DefaultConfigFlags.Namespace, "If present, the namespace scope for this CLI request")
+	cmd.Flags().StringVar(&o.FieldSelector, "field-selector", o.FieldSelector, "Selector (field query) to filter on, supports '=', '==', and '!='.(e.g. --field-selector key1=value1,key2=value2). The server only supports a limited number of field queries per type.")
+	cmd.Flags().StringVar(&o.SortBy, "sort-by", o.SortBy, "If non-empty, sort pods list using specified field. The field can be either 'cpu' or 'memory'.")
+	cmd.Flags().BoolVar(&o.PrintContainers, "containers", o.PrintContainers, "If present, print usage of containers within a pod.")
+	cmd.Flags().BoolVarP(&o.AllNamespaces, "all-namespaces", "A", o.AllNamespaces, "If present, list the requested object(s) across all namespaces. Namespace in current context is ignored even if specified with --namespace.")
+	cmd.Flags().BoolVar(&o.NoHeaders, "no-headers", o.NoHeaders, "If present, print output without headers.")
+	cmd.Flags().BoolVar(&o.UseProtocolBuffers, "use-protocol-buffers", o.UseProtocolBuffers, "Enables using protocol-buffers to access Metrics API.")
+	cmd.Flags().BoolVar(&o.Sum, "sum", o.Sum, "Print the sum of the resource usage")
+	return cmd
+}
+
+func (o *TopPodOptions) Complete(f cmdutil.Factory, cmd *cobra.Command, args []string) error {
+	var err error
+	if len(args) == 1 {
+		o.ResourceName = args[0]
+	} else if len(args) > 1 {
+		return cmdutil.UsageErrorf(cmd, "%s", cmd.Use)
+	}
+
+	o.Namespace, _, err = f.ToRawKubeConfigLoader().Namespace()
+	if err != nil {
+		return err
+	}
+	clientset, err := f.KubernetesClientSet()
+	if err != nil {
+		return err
+	}
+
+	o.DiscoveryClient = clientset.DiscoveryClient
+	config, err := f.ToRESTConfig()
+	if err != nil {
+		return err
+	}
+	if o.UseProtocolBuffers {
+		config.ContentType = "application/vnd.kubernetes.protobuf"
+	}
+	o.MetricsClient, err = metricsclientset.NewForConfig(config)
+	if err != nil {
+		return err
+	}
+
+	o.PodClient = clientset.CoreV1()
+
+	o.Printer = NewTopCmdPrinter(o.Out)
+	return nil
+}
+
+func (o *TopPodOptions) Validate() error {
+	if len(o.SortBy) > 0 {
+		if o.SortBy != sortByCPU && o.SortBy != sortByMemory {
+			return errors.New("--sort-by accepts only cpu or memory")
+		}
+	}
+	if len(o.ResourceName) > 0 && (len(o.LabelSelector) > 0 || len(o.FieldSelector) > 0) {
+		return errors.New("only one of NAME or selector can be provided")
+	}
+	return nil
+}
+
+func (o TopPodOptions) RunTopPod() error {
```

Comment:
```suggestion
func (o *TopPodOptions) RunTopPod() error {
```
How about define the method on `pointer` to address the `Goland inspection warning`:
```
Inspection info: Reports structures with methods that use a mixture of types: value and pointer receivers. Such usage is not recommended by the Go Documentation.
```
https://go.dev/doc/faq#methods_on_values_or_pointers